### PR TITLE
fix for kernels v6.3 and mitigation for tracee.pid in tests

### PR DESCRIPTION
**Pull request by rafaeldtinoco**
commit cea89a66 (HEAD -> include-v6, rafaeldtinoco/include-v6)
Author: Rafael David Tinoco <rafaeldtinoco@gmail.com>
Date:   Mon May 8 23:45:33 2023

    workflow: enable v6.1 and v6.3 tests after fix

commit f45cc1ba
Author: Rafael David Tinoco <rafaeldtinoco@gmail.com>
Date:   Mon May 8 23:42:07 2023

    ebpf: CO-RE: deal with new kernel_cap_t type

commit 57520600
Author: Rafael David Tinoco <rafaeldtinoco@gmail.com>
Date:   Mon May 8 12:54:31 2023

    workflow: fix debug-shell job name

commit 6aee7293
Author: Rafael David Tinoco <rafaeldtinoco@gmail.com>
Date:   Mon May 8 12:53:06 2023

    tests: cleanup tracee.pid in case not cleaned

----

The only change that isn't chore is the change that fixes v6.3 kernels (commit f45cc1ba) and the way to test it:

#### test case

```console
#### command 01

$ sudo ./dist/tracee --output json --filter event=com
mit_creds --filter comm=capsh

#### command 02

$ sudo capsh --drop=cap_sys_admin -- -c 'sleep 1000'

#### command 03

$ sudo cat /proc/$(pidof sleep)/status | grep ^Cap
CapInh: 0000000000000000
CapPrm: 000001ffffdfffff
CapEff: 000001ffffdfffff
CapBnd: 000001ffffdfffff
CapAmb: 0000000000000000
```

#### in a v6.3.1 kernel (vanilla upstream)
```json
{"timestamp":1683599572209837155,"threadStartTime":1683599572207592072,"processorId":12,"processId":1645572,"cgroupId":6255,"threadId":1645572,"parentProcessId":1645571,"hostProcessId":1645572,"hostThreadId":1645572,"hostParentProcessId":1645571,"userId":0,"mountNamespace":4026531841,"pidNamespace":4026531836,"processName":"capsh","hostName":"fujitsu","container":{},"kubernetes":{},"eventId":"723","eventName":"commit_creds","matchedPolicies":[""],"argsNum":2,"returnValue":0,"syscall":"prctl","stackAddresses":null,"contextFlags":{"containerStarted":false,"isCompat":false},"args":[{"name":"old_cred","type":"slim_cred_t","value":{"Uid":0,"Gid":0,"Suid":0,"Sgid":0,"Euid":0,"Egid":0,"Fsuid":0,"Fsgid":0,"UserNamespace":4026531837,"SecureBits":0,"CapInheritable":0,"CapPermitted":2199023255551,"CapEffective":2199023255551,"CapBounding":2199023255551,"CapAmbient":0}},{"name":"new_cred","type":"slim_cred_t","value":{"Uid":0,"Gid":0,"Suid":0,"Sgid":0,"Euid":0,"Egid":0,"Fsuid":0,"Fsgid":0,"UserNamespace":4026531837,"SecureBits":0,"CapInheritable":0,"CapPermitted":2199023255551,"CapEffective":2199023255551,"CapBounding":2199021158399,"CapAmbient":0}}]}
```

#### in a 5.15.0-69-generic kernel (ubuntu flavored)

```json
{"timestamp":1683599381252829883,"threadStartTime":1683599381250171936,"processorId":5,"processId":2981,"cgroupId":4098,"threadId":2981,"parentProcessId":2980,"hostProcessId":2981,"hostThreadId":2981,"hostParentProcessId":2980,"userId":0,"mountNamespace":4026531841,"pidNamespace":4026531836,"processName":"capsh","hostName":"e2etests","container":{},"kubernetes":{},"eventId":"723","eventName":"commit_creds","matchedPolicies":[""],"argsNum":2,"returnValue":0,"syscall":"prctl","stackAddresses":null,"contextFlags":{"containerStarted":false,"isCompat":false},"args":[{"name":"old_cred","type":"slim_cred_t","value":{"Uid":0,"Gid":0,"Suid":0,"Sgid":0,"Euid":0,"Egid":0,"Fsuid":0,"Fsgid":0,"UserNamespace":4026531837,"SecureBits":0,"CapInheritable":0,"CapPermitted":2199023255551,"CapEffective":2199023255551,"CapBounding":2199023255551,"CapAmbient":0}},{"name":"new_cred","type":"slim_cred_t","value":{"Uid":0,"Gid":0,"Suid":0,"Sgid":0,"Euid":0,"Egid":0,"Fsuid":0,"Fsgid":0,"UserNamespace":4026531837,"SecureBits":0,"CapInheritable":0,"CapPermitted":2199023255551,"CapEffective":2199023255551,"CapBounding":2199021158399,"CapAmbient":0}}]}
```

### CODE REVIEW

**Comment by rafaeldtinoco** [comment]
This is ready for review. The only thing having errors in v6.1 and v6.3 kernels is the e2e instrumentation tests for file_modification event (opened https://github.com/aquasecurity/tracee/issues/3077 for that issue). 

I don't see why not merging this one (if you agree) because then the tests will get fixed once that PR is merged (hopefully).

File: pkg/ebpf/c/common/capabilities.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+// store this array in one piece
+//
+// NEW NOTE: Recently, (2023), kernel has started using an u64 for all
+// capabilities, instead of using variable u32 array. Use type flavors to
+// deal with that.
+//
+
+static __always_inline u64 credcap_to_slimcap(void *from)
+{
+    kernel_cap_t___older to = {0};
```
Comment (geyslan) [inline pkg/ebpf/c/common/capabilities.h:21]: Perhaps moving this to `if (bpf_core_field_exists(to.cap)) {` scope?

**Review by geyslan — APPROVED** [review]
LGTM

File: pkg/ebpf/c/common/capabilities.h (hunk 1 above)
Comment (rafaeldtinoco) [reply pkg/ebpf/c/common/capabilities.h:21]: I also thought about it, but then, having the type flavors here keeps there cleaner (which is the "business logic" for the event)... that is why I kept it here. No biggie, both would be correct.

**Comment by rafaeldtinoco** [comment]
image

I have opened an issue for the failing tests. All the rest (in this PR) is good.

File: pkg/ebpf/c/tracee.bpf.c [hunk 2]
```diff
@@ -1964,78 +1965,75 @@ int BPF_KPROBE(trace_commit_creds)
...
+
+    slim_cred_t oslim = {0};
+    slim_cred_t nslim = {0};
+
+    struct user_namespace *userns_old = READ_KERN(ocred->user_ns);
+    struct user_namespace *userns_new = READ_KERN(ncred->user_ns);
+
+    // old credentials
+
+    oslim.uid = READ_KERN(ocred->uid.val);
```
Comment (yanivagman) [inline pkg/ebpf/c/tracee.bpf.c:1979]: Let's change back to new_ and old_, like discussed offline

**Pull request merged**